### PR TITLE
Scripts pin

### PR DESCRIPTION
**Pull request by IFX-Anusha**
- Combined make-pins.py and make-pins-csv.py into a single file for generation of the pin details from pin packages
- Replaced CSV file generation with table

-  Evaluated the possibility of defining the pin package file path in the mpconfigboard.mk but then as there are releases in the file path, updating that often will be a problem. So better that is part of the script itself.

### CODE REVIEW

File: ports/psoc6/Makefile [hunk 1]
```diff
@@ -100,19 +100,15 @@ GEN_PINS_SRC := $(BUILD)/pins_$(BOARD).c
...
 
 GENERATED_PINS = $(GEN_PINS_SRC) $(GEN_PINS_HDR) $(GEN_PINS_QSTR)
 
 $(GENERATED_PINS): 
 	@echo "Generating $@"
-	$(PYTHON) boards/make-pins-csv.py --gen-pin-for $(PIN_PACKAGE_FILE) --save-pins-csv-at $(BOARD_PINS) --save-pins-af-csv-at $(BOARD_AF_PINS)
 	$(MKDIR) -p $(BUILD)/genhdr
-	$(PYTHON) boards/make-pins.py --hdr $(GEN_PINS_HDR) --qstr $(GEN_PINS_QSTR) > $(GEN_PINS_SRC)
+	$(PYTHON) boards/make-pins.py --gen-pin-for $(PIN_PACKAGE_FILE) --hdr $(GEN_PINS_HDR) --qstr $(GEN_PINS_QSTR) > $(GEN_PINS_SRC)
 	rm -rf $(BOARD_PINS) $(BOARD_AF_PINS)
```
Comment (jaenrig-ifx) [inline ports/psoc6/Makefile:110]: The rm is no longer needed or?

**Review by jaenrig-ifx — CHANGES_REQUESTED** [review]
Nice 😃  Simpler now.
Only the thing we found out. It would be nice to find out the out of index objects there. 
If that gets difficult, we can create a new bug ticket for that...

File: ports/psoc6/Makefile (hunk 1 above)
Comment (IFX-Anusha) [reply ports/psoc6/Makefile]: Fixed

**Review by NikhitaR-IFX — APPROVED** [review]
Looks all good!

**Review by jaenrig-ifx — APPROVED** [review]

**Pull request merged**